noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/components/private_call_data_validator.nr

```diff
@@ -238,6 +232,10 @@ impl PrivateCallDataValidator {
             assert(is_own_storage, "call stack storage address does not match expected contract address");
         }
 
+        assert_eq(
+            call_context.function_selector, self.data.call_stack_item.function_data.selector, "function selector in call context does not match call stack item"
```

Comment:
I don't think we checked this before. We need to ensure that a nested call's function selector is what the caller expects.